### ISSUE TITLE
Add support for WalletConnect

### ISSUE THREAD

**Issue by CodeSandwich** [trezor/trezor-suite#3260]
**Is your feature request related to a problem? Please describe.**
I'm using a dapp which supports only WalletConnect, because its client is a standalone desktop app. I want to use it with my Trezor wallet for extra security.

**Describe the solution you'd like**
I'd like Trezor Suite web wallet to support signing messages provided to it via WalletConnect.

**Describe alternatives you've considered**
I've been looking for any wallet supporting both Trezor and WalletConnect, but there don't seem to be any options.

**Additional context**
None.

**Comment by prusnak** [comment]
it seems to me that Mobile App (https://github.com/trezor/trezor-suite/issues/2736) needs to be done first

**Comment by CodeSandwich** [comment]
Honest question: does it make sense to add WalletConnect to a wallet runnable on a desktop? It sounds reasonable to me and I have a use case which would be solved with this feature. But there seem to be 0 non-mobile wallets with WalletConnect and you're immediately thinking about the mobile version too.

**Comment by ddaddy** [comment]
+1 for WalletConnect support in the desktop suite please.

**Comment by Saladin1812** [comment]
+1 for this as I want this for Gitcoin bounties and other Dapps. Gitcoin can connect to a wallet with WalletConnect. Gitcoin itself is not on trezor but payments in ETH/ETH or others can easily be reached to the wallets compatible with WalletConnect. Currently I use metamask

**Comment by SlugFiller** [comment]
On Desktop, the WalletConnect image can be transferred via clipboard or screen capture instead of a camera. OLE drag and drop is also an option on Windows specifically (linux should have something similar as well).

WalletConnect is quickly gaining ground as a standard, due to the fact that it doesn't require any extra plugins, drivers, services, or even a modern browser.

**Comment by aza0092** [comment]
any updates on WalletConnect integration?

**Comment by shenhab** [comment]
Any update on this Feature request?

**Comment by cryptodev100** [comment]
Any updates?
It would be great if Dapps become more independent of specific hardware-wallets, therefore I believe that we need an abstraction-layer like WalletConnect.

**Comment by trymeouteh** [comment]
Would like to see this to interact with other web3 apps.

**Comment by btfdip** [comment]
super smooth when I use the integration between ledger and walletconnect, hope that trezor can also consider this feature.

**Comment by stepmes** [comment]
Thanks everyone for this input, we're noting it down and will consider it when working on future improvements.

**Comment by cryptodev100** [comment]
What I would like to see is a vision on how Trezor plans to support a large ecosystem of Dapps.
Neither can we expect that everybody uses Metamask, nor can we expect that every Dapp adds Trezor-specific code.
In fact, my user experience with Trezor and MetaMask has been extremely painful over the last months.
Many Dapps refused to work with Trezor although they have Metamask-support.
Therefore, I am already thinking about buying a Ledger just for the sake of WalletConnect-support.

I don't expect that anything is implemented right now, but I want to see a path forward where Dapps can work seamlessly instead of hoping for a miracle or hoping that the current MetaMask-version is not broken in combination with the current Trezor-firmware and Dapp.

**Comment by SlugFiller** [comment]
Is there some sort of weird policy to close an issue that has not been resolve, addressed, given so much as a workaround, or even any actual input from developers?

This issue still exists, and is a blocker for many people. Closing without so much as a single on-topic comment says "We don't care about our customers"

**Comment by cryptodev100** [comment]
> Is there some sort of weird policy to close an issue that has not been resolve, addressed, given so much as a workaround, or even any actual input from developers?
> 
> This issue still exists, and is a blocker for many people. Closing without so much as a single on-topic comment says "We don't care about our customers"

Exactly, what I would expect is some kind of roadmap for making Dapps usable.
Right now, many Dapps are either painful to use or completely unusable with Trezor.
If cofounders like @prusnak cannot give more details about the general problem of Wallet-abstraction-layers, then this doesn't shed a good light on Trezor.
I am not claiming that WalletConnect is the perfect solution for Trezor, but at least it's a working solution.

**Comment by cryptodev100** [comment]
> Honest question: does it make sense to add WalletConnect to a wallet runnable on a desktop? It sounds reasonable to me and I have a use case which would be solved with this feature. But there seem to be 0 non-mobile wallets with WalletConnect and you're immediately thinking about the mobile version too.

Over the long term, we could question the focus on Desktop and start to think mobile-first.
Nevertheless, even if we stay on the Desktop, a real problem right now is Dapp-compatibility.
Right now, I am often forced to use Trezor in combination with MetaMask, which I don't like because I faced many MetaMask-bugs over the last months.
In particular, I observed Dapps that work with standalone-MetaMask, but refused to work with MetaMask in combination with Trezor, and I observed Dapps that only work with specific combinations of MetaMask-version + Trezor-firmware-version.
Every time I try a new Dapp, it feels like a dice-roll whether the Dapp works with Trezor.

All things considered, it is clear that an over-reliance on MetaMask cannot be the endgame-solution.
Moreover, it is clear that not every Dapp can add vendor-specific code for each and every hardware-wallet that needs to be supported.
To solve all those issues, we need a wallet-abstraction-layer between Dapps and hardware-wallets, and WalletConnect is one of the first solutions that came to my mind as a wallet-abstraction-layer.

**Comment by hynek-jina** [comment]
@SlugFiller @cryptodev100
Wee need to investigate impact of this feature from many perspectives.
Like workload, impact on other components, future usability, complexity of testing, how much people need it, and so on.
Based on this evaluation we might open another issue later on. It's not on the road map now.

For roadmap see: https://github.com/orgs/trezor/projects/28#card-47974233

**Comment by SlugFiller** [comment]
@hynek-jina The expected flow, which most GitHub projects follow, is to add tag, e.g. "longterm", ask clarifying questions, if any are needed, and/or open a poll to measure how much an issue attracts attention compared to other issues.

In the case of a non-commercial project with a small team or single developer (hardly applicable to a 180EU a pop device, but we'll set that aside), a "Help wanted" or "Pull request okay" tag is used.

Closing an unresolved issue like it never existed is not a normal response.

**Comment by cryptodev100** [comment]
> @SlugFiller @cryptodev100 Wee need to investigate impact of this feature from many perspectives. Like workload, impact on other components, future usability, complexity of testing, how much people need it, and so on. Based on this evaluation we might open another issue later on. It's not on the road map now.
> 
> For roadmap see: https://github.com/orgs/trezor/projects/28#card-47974233

I understand that this is not a small feature that is implemented within a fortnight.
But ultimately, it's a fundamental question about third-party-Dapps in general.
Right now, I cannot see any plan at all for third-party-Dapp-compatibility, regardless of whether we are talking about WalletConnect or something else.
Sure, there will be some users who use only Trezor-Suite, but other users want to use third-party-Dapps as well.

**Comment by trymeouteh** [comment]
> it seems to me that Mobile App (#2736) needs to be done first

Not at all

For example [Unstoppable Domains](https://unstoppabledomains.com/manager) support WalletConnect for managing an Unstoppable Domain which you can use a mobile wallet or a desktop wallet.

By Adding WalletConnect support in the desktop app it will also complete [this feature request](https://github.com/trezor/trezor-suite/issues/3498) for Unstoppable Domains and potentially many other feature requests for Trezor Suite to integrate with other Dapps.

**Comment by cryptodev100** [comment]
> > it seems to me that Mobile App (#2736) needs to be done first
> 
> Not at all
> 
> For example [Unstoppable Domains](https://unstoppabledomains.com/manager) support WalletConnect for managing an Unstoppable Domain which you can use a mobile wallet or a desktop wallet.
> 
> By Adding WalletConnect support in the desktop app it will also complete [this feature request](https://github.com/trezor/trezor-suite/issues/3498) for Unstoppable Domains and potentially many other feature requests for Trezor Suite to integrate with other Dapps.

I agree, although WalletConnect seems to be primarily targeted for mobile wallets, I don't see a reason why it should not work for Desktop-suites.
By the way, https://github.com/trezor/trezor-suite/issues/3498 is a good example of the underlying problem:
It would be infeasible to submit a PR for almost every Dapp that needs to be supported, therefore there needs to be a generic Wallet-protocol to support most Dapps out of the box (not MetaMask).

**Comment by trymeouteh** [comment]
Wallet Connect does work with desktop wallets by copying and pasting and it would be amazing to see Trezor Suite having a QR code screen scanner which can scan QR codes in other windows on your computer.

WalletConnect v2 is decentralized and having WalletConnect added to Trezor Suite will integrate so many Dapps into Trezor.

**Comment by ffrediani** [comment]
Hi folks

I came to put a request for Wallet Connect for Trezor Suite and I'm glad someone brought it already. I want to share my view on it and how it could possibly ease things if existed. 

I came across a situation where when trying to use Trezor with most known Solana wallets there isn't full support yet on any of them(reported on this issue https://github.com/trezor/trezor-firmware/issues/4371).

If there was a WalletConnect feature in Trezor Suite it could be an option to connect to Dapps while wallets still don't integrate them with Trezor.

**Comment by Hannsek** [comment]
Hi, we are currently planning to implement this feature. ETA next year.

**Comment by surfaceflinger** [comment]
@Hannsek glad to hear that, maybe in this case it would be a good idea to reopen this to centralize discussion?

Not sure what UX you guys are planning, but people above suggested dragging the QR code which seems dumb. WalletConnect popups have an option to copy a string which then could be copied inside Trezor Suite.

Also, since you're planning to include this, please also consider implementing https://blowfish.xyz/
Internet is now plagued with drainers which AFAIK don't even look like a transaction but instead it's signing a message which might look completely harmless. Blowfish does simulations of what will happen once you sign/send a transaction and it's quite good at it. Some examples to test Blowfish in other wallets are available here https://examples.blowfish.tools/

At the same point sending transactions to Blowfish might be a privacy concern so Trezor Suite should ask user on first launch if this should be enabled or disabled.

**Comment by Hannsek** [comment]
Copying the link to trezor suite is not the best option out there. Usually, deep links are used for that.
WalletConnect itself has some kind of transaction simulation, however, we are looking into some other, more sophisticated solution as well.
Stay tuned.

**Comment by surfaceflinger** [comment]
Awesome, hyped for that :)
